### PR TITLE
fix: improve error message and performance in JS functions

### LINKED ISSUE #12179: [Feature]-[762]:bring back error message `async functions in sync fields`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

When we use and async function (e.g. Query.run()) in a non-async field (e.g. `table.tableData`)

We used to throw an explicit error stating `Cannot use async functions in sync field`.
This was removed as having this check was hindering performance.

Right now, `TypeError: Api1.run is not a function` is shown which is misleading for the users.

### Why should this be worked on?

We need to address this, so the users are not mislead.

### LINKED ISSUE #13273: [Bug]: API call inside a JS is displayed as an error when bound to a widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Calling an API/datasource query in a JS object and binding this to a widget throws an error stating `Api1.run` is not a function. Error message needs to be improved since async function is not allowed in data fields.

### Steps To Reproduce

1. Go to app at [this](https://release.app.appsmith.com/app/git-connected-app-with-js-objects/page1-62668caa01f360366e25e4d2/edit) location
2. Observe the binding on the table widget display an error stating `mockAPI.run` is not a function

image

### Public Sample App

https://release.app.appsmith.com/app/git-connected-app-with-js-objects/page1-62668caa01f360366e25e4d2/edit

### Version

Cloud

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

- Added logic to replace async function undefined error with "{{actionName}} cannot be used in this field".
- This change improves performance for 
  - ParseJSActions
  - Triggers execution
  - Each Appsmith framework action execution.
- This change adds all platform functions to evalContext permanently.

Fixes #12179
Fixes #13273

Internal discussion for error message :- https://theappsmith.slack.com/archives/C02K0SZQ7V3/p1667457021297869?thread_ts=1667385039.225229&cid=C02K0SZQ7V3

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Performance improvement

## How Has This Been Tested?

- Manual
- Jest
- Cypress

### Test Plan

- [ ] https://github.com/appsmithorg/TestSmith/issues/2086

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/ce/workers/Evaluation/dataTreeUtils.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export function makeEntityConfigsAsObjProperties(
   for (const entityName of Object.keys(dataTree)) {
     const entityConfig = Object.getPrototypeOf(dataTree[entityName]) || {};
     const entity = dataTree[entityName];
-    newDataTree[entityName] = { ...entityConfig, ...entity };
+    newDataTree[entityName] = Object.assign({}, entityConfig, entity);
```
Comment (Rishabh-Rathod) [inline app/client/src/ce/workers/Evaluation/dataTreeUtils.ts:55]: Replaced object spread by object.assign

**Comment by SatishGandham** [comment]
/perf-test
Just testing perf infra

**Review by ohansFavour — APPROVED** [review]

File: app/client/src/workers/Evaluation/evaluate.ts [hunk 2]
```diff
@@ -135,63 +147,54 @@ export const createGlobalData = (args: createGlobalDataArgs) => {
...
-    if (context.globalContext) {
-      Object.entries(context.globalContext).forEach(([key, value]) => {
-        GLOBAL_DATA[key] = value;
-      });
-    }
-  }
-  if (isTriggerBased) {
-    //// Add internal functions to dataTree;
-    const dataTreeWithFunctions = enhanceDataTreeWithFunctions(
-      dataTree,
```
Comment (Rishabh-Rathod) [inline app/client/src/workers/Evaluation/evaluate.ts:156]: enhanceDataTreeWithFunctions

**Pull request merged**